layouts/partials/helpers/fragments.html

```diff
@@ -32,49 +26,56 @@
 {{- $page_scratch.Set "fragment_directories" (slice) -}}
 {{- range $index, $section := $page_scratch.Get "sections" -}}
   {{- $directory := $section -}}
-  {{- $page_scratch.Set "active_page" ($real_page.Site.GetPage "page" (printf "%s/%s" $directory "_global")) -}}
   {{- if (eq (add $index 1) (len ($page_scratch.Get "sections"))) -}}
     {{- if ne $real_page.Kind "page" -}}
+      {{/* Last item is the page being rendered. If it's a section or homepage,
+        fragments must be at _index directory */}}
       {{- $page_scratch.Set "active_page" ($real_page.Site.GetPage "page" (printf "%s%s" $real_page.Dir "_index")) -}}
+    {{- else -}}
+      {{- $page_scratch.Set "active_page" $page -}}
     {{- end -}}
   {{- else if gt $index 1 -}}
+    {{/* Sections between the first and last are found and their _global directories
+      are taken into account */}}
     {{- $directory := (delimit (first (add $index 1) $sections) "/") -}}
     {{- $page_scratch.Set "active_page" ($real_page.Site.GetPage "page" (printf "%s/%s" $directory "_global")) -}}
+  {{- else -}}
+    {{/* First section is always the root directory and we look for it's _global
+      directory */}}
+    {{- $page_scratch.Set "active_page" ($real_page.Site.GetPage "page" (printf "%s/%s" $directory "_global")) -}}
   {{- end -}}
   {{- $page_scratch.Add "fragment_directories" ($page_scratch.Get "active_page") -}}
 {{- end -}}
 
-{{- $page_scratch.Set "global_fragments" (slice) -}}
+{{- $page_scratch.Set "fragments" (slice) -}}
 {{- range $directory := sort ($page_scratch.Get "fragment_directories") -}}
   {{- if $directory -}}
+    {{- $page_scratch.Add "fragments" . -}}
```

Comment:
Why the double add to `fragments`?

Reply:
Line 53 is adding the `index.md` and line 55 is adding the resources of said index file.